tests/test_tasks/test_spatial.py

```diff
@@ -0,0 +1,117 @@
+from typing import Any, Dict, Tuple
+
+import pytest
+
+from sentinelhub import CRS, BBox
+
+from eogrow.tasks.spatial import get_array_slices
+
+
+@pytest.mark.parametrize(
+    "input_params, expected",
+    [
+        pytest.param(
+            {
+                "bbox": BBox(((730000, 4400000), (731000, 4401000)), crs=CRS("32638")),
+                "slice_bbox": BBox(((730000, 4400000), (731000, 4401000)), crs=CRS("32638")),
+                "resolution": (10, 10),
+            },
+            (slice(0, 100, None), slice(0, 100, None)),
+            id="clean test",
+        ),
+        pytest.param(
+            {
+                "bbox": BBox(((730000, 4400000), (731000, 4401000)), crs=CRS("32638")),
+                "slice_bbox": BBox(((730000, 4400000), (730990, 4400650)), crs=CRS("32638")),
+                "resolution": (10, 10),
+            },
+            (slice(0, 99, None), slice(0, 65, None)),
+            id="less regular test",
+        ),
+        pytest.param(
+            {
+                "bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "slice_bbox": BBox(((730000, 4400000), (732000, 4401000)), crs=CRS("32638")),
+                "resolution": (10, 10),
+            },
+            (slice(0, 200, None), slice(0, 100, None)),
+            id="random test",
```

Comment:
doesnt this one match the first test? except it's not symmetrical :thinking: 
I would keep this one and throw the first one out since they both test the `fully covered` test case

Reply:
updated